src/parser/warrior/arms/modules/spells/azeritetraits/TestOfMight.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import Analyzer from 'parser/core/Analyzer';
+import Enemies from 'parser/shared/modules/Enemies';
+import { formatPercentage, formatNumber } from 'common/format';
+import { calculateAzeriteEffects } from 'common/stats';
+import SPELLS from 'common/SPELLS';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'interface/others/TraitStatisticBox';
+
+const RAGE_PER_PROC = 10;
+
+const testOfMightStats = traits => Object.values(traits).reduce((obj, rank) => {
+    const [strength] = calculateAzeriteEffects(SPELLS.TEST_OF_MIGHT.id, rank);
+    obj.strength += strength;
+    obj.traits += 1;
+    return obj;
+}, {
+    strength: 0,
+    traits: 0,
+});
+
+export const STAT_TRACKER = {
+    strength: combatant => {
+        return testOfMightStats(combatant.traitsBySpellId[SPELLS.TEST_OF_MIGHT.id]).strength;
+    },
+};
+
+/**
+ * Normal:
+ * When Colossus Smash expires, your Strength is increased by 29 for every 10 Rage you spent during Colossus Smash. Lasts 12 sec.
+ * 
+ * Warbreaker:
+ * When Warbreaker expires, your Strength is increased by 29 for every 10 Rage you spent during Warbreaker. Lasts 12 sec.
+ */
+
+class TestOfMight extends Analyzer {
+
+    static dependencies = {
+        enemies: Enemies,
+    };
+
+    strength = 0;
+    traits = 0;
+
+    proc = 0;
+    strengthOnToM = 0;
+    totalStrength = 0;
+    rageSpendDuringDebuff = 0;
+
+    constructor(...args) {
+        super(...args);
+        this.active = this.selectedCombatant.hasTrait(SPELLS.TEST_OF_MIGHT.id);
+        if(!this.active) {
+            return;
+        }
+
+        const { strength, traits } = testOfMightStats(this.selectedCombatant.traitsBySpellId[SPELLS.TEST_OF_MIGHT.id]);
+        this.strength = strength;
+        this.traits = traits;
+    }
+
+    on_byPlayer_applybuff(event) {
+        if (event.ability.guid !== SPELLS.TEST_OF_MIGHT_BUFF.id) {
+            return;
+        }
+        this.proc += 1;
+        
+        this.strengthOnToM += this.strength * (this.rageSpendDuringDebuff / RAGE_PER_PROC);
+        this.totalStrength += this.strengthOnToM;
+
+        this.rageSpendDuringDebuff = 0;
+        this.strengthOnToM = 0;
+    }
+
+    on_byPlayer_cast(event) {
+        if (!event.classResources ||
```

Comment:
Maybe this could be simplified to something like (I haven't checked, not 100% sure)
```javascript
const rage = event.classResources.find(e => e.type === RESOURCE_TYPES.RAGE.id);
if (!rage) { 
    return;
}
....
    this.rageSpendDuringDebuff += rage.cost / 10;
```
Cast events *should* always have a `classResources` field, and if `find()` can't find anything, it returns `undefined`.